libraries/AP_Radio/AP_Radio.cpp

```diff
@@ -0,0 +1,254 @@
+    // @Range: 0 40
+    // @User: Advanced
+    AP_GROUPINFO("_BZOFS",  15, AP_Radio, tx_buzzer_adjust, 25),
+
+    // @Param: _ABTIME
+    // @DisplayName: Auto-bind time
+    // @Description: When non-zero this sets the time with no transmitter packets before we start looking for auto-bind packets.
+    // @Range: 0 120
+    // @User: Advanced
+    AP_GROUPINFO("_ABTIME",  16, AP_Radio, auto_bind_time, 0),
+
+    // @Param: _ABLVL
+    // @DisplayName: Auto-bind level
+    // @Description: This sets the minimum RSSI of an auto-bind packet for it to be accepted. This should be set so that auto-bind will only happen at short range to minimise the change of an auto-bind happening accidentially
+    // @Range: 0 31
+    // @User: Advanced
+    AP_GROUPINFO("_ABLVL",  17, AP_Radio, auto_bind_rssi, 0),
+    
+    AP_GROUPEND
+};
+
+AP_Radio *AP_Radio::_instance;
+
+// constructor
+AP_Radio::AP_Radio(void)
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    if (_instance != nullptr) {
+        AP_HAL::panic("Multiple AP_Radio declarations");
+    }
+    _instance = this;
+}
+
+bool AP_Radio::init(void)
+{
+    switch (radio_type) {
+    case RADIO_TYPE_CYRF6936:
+        driver = new AP_Radio_cypress(*this);
+        break;
+    case RADIO_TYPE_CC2500:
+        driver = new AP_Radio_cc2500(*this);
+        break;
+    default:
+        break;
+    }
+    if (!driver) {
+        return false;
+    }
+    return driver->init();
+}
+
+bool AP_Radio::reset(void)
+{
+    if (!driver) {
+        return false;
+    }
+    return driver->reset();
+}
+
+bool AP_Radio::send(const uint8_t *pkt, uint16_t len)
+{
+    if (!driver) {
+        return false;
+    }
+    return driver->send(pkt, len);
+}
+
+void AP_Radio::start_recv_bind(void)
+{
+    if (!driver) {
+        return;
+    }
+    return driver->start_recv_bind();
+}
+
+const AP_Radio::stats &AP_Radio::get_stats(void)
+{
+    return driver->get_stats();
+}
+
+uint8_t AP_Radio::num_channels(void)
+{
+    if (!driver) {
+        return 0;
+    }
+    return driver->num_channels();
+}
+
+uint16_t AP_Radio::read(uint8_t chan)
+{
+    if (!driver) {
+        return 0;
+    }
+    return driver->read(chan);
+}
+
+uint32_t AP_Radio::last_recv_us(void)
+{
+    if (!driver) {
+        return 0;
+    }
+    return driver->last_recv_us();
+}
+
+// handle a data96 mavlink packet for fw upload
+void AP_Radio::handle_data_packet(mavlink_channel_t chan, const mavlink_data96_t &m)
+{
+    if (driver) {
+        driver->handle_data_packet(chan, m);
+    }
+}
+
+// update status, should be called from main thread
+void AP_Radio::update(void)
+{
+    if (driver) {
+        driver->update();
+    }
+}
+
+// get transmitter firmware version
+uint32_t AP_Radio::get_tx_version(void)
+{
+    if (driver) {
+        return driver->get_tx_version();
+    }
+    return 0;
+}
+
+// set the 2.4GHz wifi channel used by companion computer, so it can be avoided
+void AP_Radio::set_wifi_channel(uint8_t channel)
```

Comment:
this is more of a "avoid_wifi_channel" or "set_wifi_channel_to_avoid"  than "set_wifi_channel", causual observer/s might think this function *uses* the wifi channel.